### ISSUE TITLE
Add more normalsets

### ISSUE THREAD

**Issue by Choiyeonhyo** [ultralytics/ultralytics#20294]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

Hello ultralytics!

I am currently training and testing the Yolov8 model. All the performance indicators of the training are normal. However, the only drawback is that many FPs are detected. 
So, I want to increase the coverage of the normal dataset (without any class objects). 
However, since the background portion you suggest is recommended **not to exceed 10%**, I wonder how I can train more normal datasets. 
In my case, the 'Class 0' dataset has 400,000 images, and if I train only 10% of the 400,000 images you recommend, that is, 40,000 images, many false positives are detected. Fortunately, the objects are found well. Is there a good way? For example, 
1. How to train more normal datasets than the Class 0 dataset 
2. Or, replace the loss function

I always appreciate your great work!

**Comment by glenn-jocher** [comment]
Hi @Choiyeonhyo,

Thanks for using Ultralytics YOLO! The 10% guideline for background images is indeed flexible - it's a starting point rather than a strict rule. Here are some approaches to tackle your false positive issue:

1. You can gradually increase the proportion of normal/background images in your training set beyond 10% while monitoring validation metrics. Try increments like 15%, 20%, etc., and observe how precision and recall are affected.

2. Rather than modifying the loss function, you could:
   - Adjust confidence thresholds during inference to filter out low-confidence detections
   - Implement hard negative mining to focus training on difficult background examples
   - Use dataset splitting techniques like our `split_classify_dataset()` function to create a more balanced training set
   - Apply data augmentation to your positive samples to increase their effective number

If you decide to use more background images, ensure you're still maintaining reasonable class balance to avoid biasing the model too much toward the background class.

Let me know which approach you'd like to explore further, and I can provide more specific guidance.

**Comment by Choiyeonhyo** [comment]
> thanks for your reply!?
i have some questions for your reply
1. Is there any way to implement 'hard negative mining' in yolov8? or have to do myself?
2.  i could not find 'split_classify_dataset ' in yolov8 codes. where is?
3. i think i will try to increase more background images more over 10%, 20%.. up to 100% first!

**Comment by glenn-jocher** [comment]
Hi @Choiyeonhyo,

Regarding your questions:

1. Hard negative mining isn't built directly into YOLOv8. You would need to implement this yourself by identifying false positives from validation runs and adding them to your training set.

2. The `split_classify_dataset()` function is located in `ultralytics/data/split.py`. You can use it with:
```python
from ultralytics.data.split import split_classify_dataset
split_classify_dataset("path/to/dataset", train_ratio=0.8)
```

3. I recommend gradually increasing background images (15%, 20%, 30%) while monitoring validation metrics closely. Going to 100% background would create severe class imbalance and likely harm detection performance. The sweet spot is usually somewhere between 20-40% depending on your specific dataset.

Let me know how your experiments go!
